src/components/SocialMedia/social.spec.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { userToken } from '../../../__mocks__/mockData';
+import SocialMediaButtons from './SocialMediaButtons';
+import SocialMediaLogin from './SocialMediaLogin';
+
+const content = {
+  authReducer: {
+    isAuthenticated: true
+  }
+};
+const store = {
+  getState: () => {
+    return { ...content };
+  },
+  subscribe: () => {
+    return {};
+  },
+  dispatch: () => {
+    return {};
+  }
+};
+
+describe('SocialMediaButtons', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount(<SocialMediaButtons />);
+  });
+  it('should contain a div with the class social', () => {
+    expect(wrapper.find('div').hasClass('social'));
+  });
+  it('should contain 3 anchor links', () => {
+    const anchorTag = wrapper.find('.social').props().children;
+    expect(wrapper.find('a')).toBeTruthy();
+    expect(anchorTag.length).toEqual(3);
+  });
+});
+
+describe('SocialMediaLogin Test', () => {
+  let wrapper, location;
+  beforeEach(() => {
+    location = {
+      search: `token=${userToken}`,
+      url: '/home'
+    };
+  });
+  it('should redirect to the home page if the url was not provided', () => {
+    const { authReducer } = content;
+    delete location.url;
+    authReducer.isAuthenticated = true;
+    wrapper = mount(
+      <Provider store={store}>
+        <Router>
+          <Route
+            render={() => {
+              return <SocialMediaLogin location={location} />;
+            }}
+          />
+        </Router>
+      </Provider>
+    );
```

Comment:
```js
const store = createStore({...authReducer, isAuthenticated: true })
wrapper = mount(
      <Provider store={store}>
     ....
```

Reply:
Consider having a factory function that creates the store instead of directly mutating global data.

Reply:
Okay, we will consider that. Thank you